crates/core/tedge/src/main.rs

```diff
@@ -3,37 +3,86 @@
 
 use anyhow::Context;
 use clap::Parser;
+use std::future::Future;
+use std::path::PathBuf;
 use tedge::command::BuildCommand;
 use tedge::command::BuildContext;
+use tedge::Component;
+use tedge::TEdgeOptMulticall;
+use tedge_apt_plugin::AptCli;
 use tedge_config::system_services::set_log_level;
-use tracing::log::warn;
 
 fn main() -> anyhow::Result<()> {
-    set_log_level(tracing::Level::WARN);
+    let executable_name = executable_name();
 
-    let opt = tedge::cli::Opt::parse();
-
-    if opt.init {
-        warn!("This --init option has been deprecated and will be removed in a future release. Use the `tedge init` command instead");
-        return Ok(());
+    if executable_name.as_deref() == Some("apt") {
```

Comment:
Can you change it to also look for the full name of the binary, `tedge-apt-plugin` as well?

For example (obviously modify the code, but you get the idea):

```
if executable_name.as_deref() == Some("apt") || executable_name.as_deref() == Some("tedge-apt-plugin") {
```

Then for the symlink creation, I would create the symlinks under:
* `/usr/bin/tedge` -> `/usr/bin/tedge-apt-plugin` (we can't use /usr/bin/apt as this is already reserved by the real apt binary)

Then the actual tedge-apt-plugin package can handle the creation of the `/etc/tedge/sm-plugins/apt` symlink when the package is installed.

Reply:
Ah sure, I overlooked that

Reply:
Looks good now.